File: fe-admin/src/app/core/model/request/search-patient-request.ts
import {PaginationFormat} from "./pagination-format";

export interface SearchPatientRequest extends PaginationFormat{
    sortBy:string;
    codeKeyword:string;
    nameKeyword:string;
    phoneKeyword:string;
    guardianNameKeyword:string;
    guardianPhoneKeyword:string;
}

File: be-api/src/main/java/cd/beapi/dto/request/SearchPatientRequest.java
package cd.beapi.dto.request;

import cd.beapi.enumerate.PatientSortOption;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.experimental.FieldDefaults;

@Getter
@FieldDefaults(level = AccessLevel.PRIVATE)
public class SearchPatientRequest extends PaginationFormat {
    PatientSortOption sortBy;
    String codeKeyword;
    String nameKeyword;
    String phoneKeyword;
    String guardianNameKeyword;
    String guardianPhoneKeyword;
}
